### PR TITLE
Feature/map markers

### PR DESCRIPTION
**Pull request by EspressoCode**
Markers added to all the buildings. A small information window is shown on top of each marker when the marker is tapped and the markers dynamically get removed if one zooms in so close, to be more precise, that is because we do not like to see the marker during indoor navigation.

### CODE REVIEW

File: lib/widget/map.dart [hunk 1]
```diff
@@ -103,12 +106,19 @@ class _MapState extends State<Map> {
...
           polygons: buildingHighlights,
           polylines: direction,
+          markers: markers,
           initialCameraPosition: _cameraPosition,
           onMapCreated: (GoogleMapController controller) {
             _completer.complete(controller);
           },
           onCameraMove: (CameraPosition cameraPosition) async {
             GoogleMapController _mapController = await _completer.future;
+            //removing markers if the zoom is greater than 20
```
Comment (DiveshPatel) [inline lib/widget/map.dart]: Should this require a setState(){}?

Comment (EspressoCode) [reply lib/widget/map.dart]: it works fine without setState(), would you please run it on your device to double check?

**Comment by EspressoCode** [comment]
> Please resolve the merge conflicts

Sorry I didn't see that, that's done now.

**Comment by EspressoCode** [comment]
I understand, but these markers are going to be there everytime only for the buildings, this will cover:

**Comment by EspressoCode** [comment]
I understand, but this will not replace the search result marker, as we talked I think we decided to have the markers for the building all the time, to cover this 2 stories:

US-11: As a student, I want to get basic pop-up information on campus buildings so that I can find further details.

US-13: As a student, I want to enter my start and end buildings using touch input on the campus map.

The marker for the search results will have to be implemented for the next step, so this PR is not meant to replace those markers.

**Comment by MrJCipherButtles** [comment]
This implementation does not close US-11 alone as discussed. You still need to collaborate for the actual pop-up to appear and that should only put a singular marker anyways which I believe was also discussed? By the way, US-13 is not part of this sprint's backlog, and it was also discussed that this was an optional user story that we were most likely not going to do within our 4 sprint time frame. I don't know why it wasn't updated in the documentation, but I have done so now. My apologies for any confusion.

**Comment by MrJCipherButtles** [comment]
@EspressoCode this PR should focus on the search result marker. Branch off this branch to save your work, then change this branch to show the search result marker. I think we need to have a team discussion on this whole marker business to be all on the same page. Because looking at both this PR and PR #86, we end up with a marker mess...

File: lib/widget/map.dart [hunk 2]
```diff
@@ -136,6 +145,13 @@ class _MapState extends State<Map> {
                         builder: (context) => GotoPage(
                           _position,
                           confirmDirection: (List<Coordinate> location) => {
+                            // setState(markers.clear(),
```
Comment (DiveshPatel) [inline lib/widget/map.dart]: Is the commented code necessary?

**Comment by EspressoCode** [comment]
ok sounds good, I apologize for not understanding the problem well, I will close this pr for now and work on adding markers based on search results only:-)

File: lib/widget/map.dart (hunk 1 above)
Comment (DiveshPatel) [reply lib/widget/map.dart]: yea it works without setState

**Comment by EspressoCode** [comment]
For a reason the changes by my latest commit have not been reflected. I close and reopen the pr to maybe go around this, as suggested by this: https://confluence.atlassian.com/bitbucketserverkb/pull-request-cannot-be-merged-or-not-showing-latest-commits-877338546.html

**Comment by EspressoCode** [comment]
With new version, a single marker is being added based on the search result.

File: lib/model/coordinate.dart [hunk 3]
```diff
@@ -5,21 +5,47 @@ import 'package:google_maps_flutter/google_maps_flutter.dart';
...
-  final String _floor;
-  final String _building;
-  final String _campus;
+  String _floor;
+  String _building;
+  String _campus;
+
+  //attributes to be used for google places search results
+  String _gPlaceTitle;
+  //String _gPlacePhotos;
```
Comment (ChesterLY) [inline lib/model/coordinate.dart]: At the moment, if it is uneeded, you can remove this comment

File: lib/model/coordinate.dart [hunk 4]
```diff
@@ -5,21 +5,47 @@ import 'package:google_maps_flutter/google_maps_flutter.dart';
...
+  Coordinate(
+    this._lat,
+    this._lng,
+    this._floor,
+    this._building,
+    this._campus, {
+    type,
+    adjCoordinates,
+    gPlaceTitle,
+    //String  gPlacePhoto,
```
Comment (ChesterLY) [inline lib/model/coordinate.dart]: Here too

File: lib/model/coordinate.dart [hunk 5]
```diff
@@ -5,21 +5,47 @@ import 'package:google_maps_flutter/google_maps_flutter.dart';
...
+    gPlaceWebsite,
+  }) {
+        _type = type;
     if (adjCoordinates != null) {
       _adjCoordinates = adjCoordinates;
       for (var adjCoordinate in _adjCoordinates) {
         adjCoordinate.addAdjCoordinate(this);
       }
     }
+     _gPlaceTitle = gPlaceTitle;
```
Comment (ChesterLY) [inline lib/model/coordinate.dart]: The formatting is a bit off, make sure you use the flutter format on every document you worked on before commiting/PR

Comment (EspressoCode) [reply lib/model/coordinate.dart]: Thanks for your comment, I made required changes. :-)

File: lib/model/coordinate.dart (hunk 3 above)
Comment (EspressoCode) [reply lib/model/coordinate.dart]: That's true, I removed it.

File: lib/model/coordinate.dart (hunk 4 above)
Comment (EspressoCode) [reply lib/model/coordinate.dart]: That's true, I removed it.

**Comment by EspressoCode** [comment]
> You're reintroducing a lot of old code back into master. You need to include your updates using the new code

Would you please check the updated code and approve if it has no problem?

**Comment by MrJCipherButtles** [comment]
@raffiJan I changed the base of this PR to be merged with your branch, if possible. Else, feel free to reject and close this PR without merging, if you prefer doing your own solution for map markers and your popup info

**Comment by DiveshPatel** [comment]
This PR will be closed as the fix in implemented in PR #145.

**Pull request closed without merging**